File: LeetCode/algorithms/easy/leetcode_242_ValidAnagram/solution.py

def isAnagram(s: str, t: str) -> bool:
    if len(s) != len(t):
        return False
    countS, countT = {}, {}

    for x in range(len(s)):
        countS[s[x]] = countS.get(s[x], 0) + 1
        countT[t[x]] = countT.get(t[x], 0) + 1
    for c in countS:
        if countS[c] != countT.get(c, 0):
            return False
    return True


print(isAnagram("anagram", "nagaram"))

File: LeetCode/algorithms/easy/leetcode_242_ValidAnagram/solution.rb
# require 'debug'

def isAnagram(s, t)
    # debugger
    return false unless s.length == t.length
    hashS = Hash.new(0)
    hashT = Hash.new(0)
    for i in 0...s.length
        hashS[s[i]] += 1
        hashT[t[i]] += 1
    end
    hashS.each{ |key, value| return false unless hashS[key] == hashT[key] }
    true
end

puts isAnagram("anagram", "nagaram")